idaes/core/util/tables.py

```diff
@@ -234,8 +235,19 @@ def create_stream_table_dataframe(
             for i in disp_dict[k]:
                 if i is None:
                     stream_attributes[key][k] = value(disp_dict[k][i])
+                    if k not in full_keys:
+                        full_keys.append(k)
                 else:
                     stream_attributes[key][k + " " + str(i)] = value(disp_dict[k][i])
+                    if k + " " + str(i) not in full_keys:
+                        full_keys.append(k + " " + str(i))
```

Comment:
Doesn't really matter but you could also do ``f"{k} {i}"`` looks slightly neater and may be slightly faster.